modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/DeployVariableValidator.java

```diff
@@ -204,129 +129,65 @@ public Map<VariableValidator, String> getValidatorMap(String variableName, Strin
      */
     public boolean isVariableValid(List<DeployVariable> deployVariables,
                                    Map<String, String> deployProperty) {
+        if (CollectionUtils.isEmpty(deployVariables) || CollectionUtils.isEmpty(deployProperty)) {
+            return true;
+        }
 
-        if (!CollectionUtils.isEmpty(deployVariables) && Objects.nonNull(deployProperty)) {
-            // check required keys
-            Set<String> userPutKeys = deployProperty.keySet();
-            Set<String> requiredKeys = getRequiredKeySet(deployVariables);
-            Set<String> ignoredKeys = new HashSet<>();
-            if (!userPutKeys.containsAll(requiredKeys)) {
-                for (String requiredKey : requiredKeys) {
-                    if (!userPutKeys.contains(requiredKey)) {
-                        ignoredKeys.add(requiredKey);
-                    }
-                }
-            }
-            if (!ignoredKeys.isEmpty()) {
-                throw new InvalidDeploymentVariableException(
-                        String.format("Required keys %s  of deploy "
-                                + " variables not found in deploy property", ignoredKeys));
-            }
-            // check input value of variables is valid by validator of Ocl context.
-            Map<String, Map<VariableValidator, String>> variableInfoMap = getVariableValidatorMap(
-                    deployVariables);
-            if (variableInfoMap.isEmpty() || userPutKeys.isEmpty()) {
+        JsonSchemaObject jsonObjectSchema = buildJsonObjectSchema(deployVariables);
```

Comment:
write a new spring bean -> ServiceVariablesJsonSchemaGenerator and move the logic there.